tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -217,6 +230,33 @@ impl ConnectionWorkersScheduler {
         cancel: CancellationToken,
     ) -> Result<Self, ConnectionWorkersSchedulerError> {
         let endpoint = Self::setup_endpoint(bind, stake_identity)?;
+
+        let connection_gen = Arc::new(AtomicU64::new(0));
+        tokio::spawn({
+            let cancel = cancel.clone();
+            let connection_gen = connection_gen.clone();
+            let mut endpoint = endpoint.clone();
+            async move {
+                loop {
+                    tokio::select! {
+                        _ = update_certificate_receiver.changed() => {
+                            let stake_identity = update_certificate_receiver.borrow_and_update();
```

Comment:
I must say, I am not sure, I understand why do you create a separate task for waiting for the `update_certificate_receiver` messages.

You already have a message loop below, where you are waiting for the transaction batches.

Why not add the `update_certificate_receiver.changed()` in there?
And when you get an update, just flush the worker cache.
This would remove the need for the generation counter.

And you update your local `endpoint` that is used to construct any new workers, when you receive an update from the `update_certificate_receiver`.
So, it seems pretty easy and matching the existing logic.

Reply:
Hm, maybe you are right -- if I don't run updates of certificate in a separate task, it cannot happen that while doing updates I start broadcasting some incoming packets.  The only a priori drawback that in this case while I will be flushing the incoming queue may grow a bit but since this is a rare event, it might be good enough.